File: 2023-12-21/1068.py
n, parentList, k = int(input()), list(map(int, input().split())), int(input())
cnt = 0

def dfs(num, list):
    list[num] = None
    for i in range(len(list)):
        if num == list[i]:
            dfs(i, list)

dfs(k, parentList)

for i in range(len(parentList)):
    if parentList[i] != None and i not in parentList:
        cnt += 1
print(cnt)

File: 2023-12-21/1068.js
const fs = require('fs');
const filePath = process.platform === 'linux' ? '/dev/stdin' : __dirname + '/input.txt';
const [L, data, K] = fs.readFileSync(filePath).toString().trim().split('\n');

const solution = (parentList, k) => {
  let cnt = 0;

  const dfs = (cutNum, list) => {
    list[cutNum] = null;

    for (let i = 0; i < list.length; i++) {
      if (cutNum === list[i]) {
        dfs(i, list);
      }
    }
  };

  dfs(k, parentList);

  for (let i = 0; i < parentList.length; i++) {
    if (parentList[i] != null && !parentList.includes(i)) {
      cnt++;
    }
  }
  console.log(cnt);
};

solution(data.split(' ').map(Number), Number(K));
